File: Codigo proyecto 2/segundo_nucleo.py
# segundo_nucleo.py
import _thread
import time

class Segundo_nucleo:
    def __init__(self, maquina):
        self.maquina = maquina

    def yaw_task(self):
        while True:
            # Leer el giroscopio y calcular el Yaw
            gyro_z = self.maquina.giro_acel.gyro.z # Ajustar según la biblioteca MPU6050
            # Suponiendo que la clase MPU6050 tiene un método para calcular el Yaw
            # self.maquina.giro_acel.calcular_yaw(gyro_z)
            time.sleep_ms(1)  # Esperar 1 ms

    def iniciar(self):
        print("Iniciando segundo núcleo...")
        _thread.start_new_thread(self.yaw_task, ())
        print("Segundo núcleo iniciado")

File: proyecto 2 micropython/segundo_nucleo.cpp
#include <freertos/FreeRTOS.h>
#include <freertos/task.h>
//#include "maquina.h"
#include "segundo_nucleo.h"
#include <Arduino.h>

Segundo_nucleo::Segundo_nucleo()
{
    
}

void Segundo_nucleo::yawTask(void *pvParameters) {
    // Convertir el parámetro al tipo Maquina*
    Maquina* maquina = (Maquina*)pvParameters;
    while(true) {
        // Leer el giroscopio y calcular el Yaw
        int16_t gyroZ = maquina->giro_acel.leerGiroscopio();
        maquina->giro_acel.calcularYaw(gyroZ);
        // Esperar 1 ms
        vTaskDelay(1 / portTICK_PERIOD_MS); // Mínima espera
    }
}

void Segundo_nucleo::iniciar(Maquina& maquina) {
    Serial.println("Iniciando segundo núcleo...");
    maquinaPtr = &maquina;  // Guardar el puntero a Maquina
    xTaskCreatePinnedToCore(
        Segundo_nucleo::yawTask, // Función a ejecutar
        "yawTask", // Nombre de la tarea
        10000, // Tamaño de la pila
        (void*)maquinaPtr, // Parámetro
        1, // Prioridad
        NULL, // Handle
        1); // Núcleo
    Serial.println("Segundo núcleo iniciado");
}
